packages/adaptive_scaffold/lib/src/adaptive_layout.dart

```diff
@@ -0,0 +1,513 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+import 'package:flutter/widgets.dart';
+import 'breakpoint.dart';
+import 'slot_layout.dart';
+import 'slot_layout_config.dart';
+
+/// A parent Widget takes in multiple [SlotLayout] components and places them
+/// into their appropriate positions on the screen.
+
+class AdaptiveLayout extends StatefulWidget {
+  /// Creates an [AdaptiveLayout] widget.
+  const AdaptiveLayout({
+    this.primaryNavigation,
+    this.secondaryNavigation,
+    this.topNavigation,
+    this.bottomNavigation,
+    this.body,
+    this.secondaryBody,
+    this.bodyRatio,
+    this.internalAnimations = true,
+    this.horizontalBody = true,
+    super.key,
+  });
+
+  /// The slot placed on the beginning side of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by any other method.
+  final SlotLayout? primaryNavigation;
+
+  /// The slot placed on the end side of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by any other method.
+  final SlotLayout? secondaryNavigation;
+
+  /// The slot placed on the top part of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by any other method.
+  final SlotLayout? topNavigation;
+
+  /// The slot placed on the bottom part of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by any other method.
+  final SlotLayout? bottomNavigation;
+
+  /// The slot that fills the rest of the space in the center.
+  final SlotLayout? body;
+
+  /// A supporting slot for [body]. Has a sliding entrance animation by default.
+  /// The default ratio for the split between [body] and [secondaryBody] is so
+  /// that the split axis is in the center of the screen.
+  final SlotLayout? secondaryBody;
+
+  /// Defines the fractional ratio of [body] to the [secondaryBody].
+  ///
+  /// For example 1 / 3 would mean [body] takes up 1/3 of the available space
```

Comment:
```suggestion
  /// For example 1/3 would mean [body] takes up 1/3 of the available space
```

You might also use decimals (0.3 and "30%" instead of 1/3), since the fraction might not render great in the API docs.